src/box/box.cc

```diff
@@ -3499,6 +3515,28 @@ bootstrap_from_master(struct replica *master)
 	/* Make the initial checkpoint */
 	if (gc_checkpoint() != 0)
 		panic("failed to create a checkpoint");
+
+	return 0;
+}
+
+/**
+ * If bootstrap failed from remote master rebootstrap it,
+ * in the case of non-critical error.
+ */
+static void
+rebootstrap_from_master(struct replica *master)
+{
+	do {
+		box_restart_replication();
```

Comment:
It's not clear why you reconnect all the appliers between the bootstrap attemts.
This is done to re-fetch remote master ballots so that `replicaset_find_join_master` can operate on fresh data.
Please, add a comment.

Reply:
I wrote comment